azavea/raster-foundry
# Use Band model, resolution, and metadata file on import

## Overview

Models for the importer had become stale relative to models in the scala
backend, so imports didn't work for a variety of reasons. This commit
modifies the whole import process from `models/*.py` to the upload
scripts and modified the scala backend to deal with the updated incoming
data.

### Summary of changes:

- synced fields between `x.scala` in the datamodel and `x.py` in
  importer
- updated scene insertion logic to deal with presence of images with
  related
- updated the style guide to include style for "messy" joins
- included geotrellis packages in `app-backend/app/build.sbt`

### Checklist

  Checking some information in the scala backend led to inspecting a complicated join, and complicated join style was added to the styleguide as a result.

## Testing Instructions

 * `vagrant up`, `vagrant ssh`, `./scripts/server | grep app-server` <-- to show errors posting if things go wrong
* `docker-compose run airflow-scheduler bash`
* `airflow run find_landsat8_scenes import_new_landsat8_scenes 2016-11-03T00:00:00` (or some other recent date just so it doesn't take too long)
* open up the airflow ui at `localhost:8080`, click on `find_landsat8_scenes`, click on the task, and click `View log`
* search for "csv rows to create scenes from" and observe how many (740 if you run for November 3)
* Wait a bit, then `./scripts/psql`, `select datasource, count(*) from scenes group by datasource`, observe that you have the right number of scenes
* edit `app-tasks/dags/sentinel2/find_sentinel2_scenes.py` -- add a `[:1]` to `tilepaths = ...` (l59-61). The `app-server` or `nginx` docker container doesn't love multiple inbound requests, and the way we have this task set up right now that's unavoidable unless there's just one tile in the list. `¯\_(ツ)_/¯`
* `airflow run find_sentinel2_scenes find_new_sentinel2_scenes 2016-11-03T00:00:00` (or some other recent date)
* Repeat the checks from landsat 8, subbing in "products to create tiles from" for "csv rows to create scenes from"

Connects #582, closes #623
